hbase_sql/src/main/java/com/splicemachine/derby/lifecycle/CostChoosingDataSetProcessorFactory.java

```diff
@@ -75,23 +77,10 @@ public DataSetProcessor chooseProcessor(@Nullable Activation activation,@Nullabl
         if (op.isOlapServer())
             return new SparkDataSetProcessor();
 
-        switch(activation.getLanguageConnectionContext().getDataSetProcessorType()){
-            case FORCED_CONTROL:
-                return new ControlDataSetProcessor(driver.getTxnSupplier(), driver.getTransactor(), driver.getOperationFactory());
-            case FORCED_SPARK:
-                return new SparkDataSetProcessor();
-            default:
-                break;
-        }
-        switch (((BaseActivation)activation).datasetProcessorType()) {
-            case FORCED_SPARK:
-            case SPARK:
-                return new SparkDataSetProcessor();
-            case FORCED_CONTROL:
-                return new ControlDataSetProcessor(driver.getTxnSupplier(), driver.getTransactor(), driver.getOperationFactory());
-            case DEFAULT_CONTROL:
-            default:
-                return new ControlDataSetProcessor(driver.getTxnSupplier(), driver.getTransactor(), driver.getOperationFactory());
+        if (((BaseActivation)activation).datasetProcessorType().isSpark()) {
```

Comment:
If we skip the check of dataSetProcessorType() in LanguageConnectionContext, we may not be able to honor the session level useSpark hint for some special cases. For example:
```
splice> connect 'jdbc:splice://localhost:1527/splicedb;user=splice;password=admin;useSpark=true' as spark_con;
explain select name from new com.splicemachine.derby.vti.SchemaFilterVTI() as b (NAME VARCHAR(128));
```
The explain will show Spark, but the query is actually sent to control. This is because we didn't call ActivationClassBuilder.setDataSetProcessorType() for this case. As a general solution, maybe we should calll this function in ScrollInsensitiveResultSetNode.generate().

Reply:
Better yet, we can remove all particular calls to ActivationClassBuilder.setDataSetProcessorType() (since they were all the same) and call it only once after the creation of ActivationClassBuilder. As we now set compilerContext.dataSetProcessorType in stone during opt phase, that will suffice.